x/mm2/keeper/msg_server.go

```diff
@@ -0,0 +1,136 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/skip-mev/slinky/x/mm2/types"
+)
+
+// msgServer is the default implementation of the x/marketmap MsgService.
+type msgServer struct {
+	k *Keeper
+}
+
+// NewMsgServer returns the default implementation of the x/marketmap message service.
+func NewMsgServer(k *Keeper) types.MsgServer {
+	return &msgServer{k}
+}
+
+var _ types.MsgServer = (*msgServer)(nil)
+
+// CreateMarkets updates the marketmap by creating markets from the given message.  All updates are made to the market
+// map and then the resulting final state is checked to verify that the end state is valid.
+func (ms msgServer) CreateMarkets(goCtx context.Context, msg *types.MsgCreateMarkets) (*types.MsgCreateMarketsResponse, error) {
+	if msg == nil {
+		return nil, fmt.Errorf("unable to process nil msg")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	params, err := ms.k.GetParams(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get marketmap params: %w", err)
+	}
+
+	found := checkMarketAuthority(msg.Authority, params)
+	if !found {
+		return nil, fmt.Errorf("request signer %s does not match module market authorities", msg.Authority)
+	}
```

Comment:
Just a sanity check (im sure ive asked this before), but are we certain that signers can't just set this to an authority address when they create the transaction but sign with a different key?

Reply:
no, we have the proto-annotation that this must be the signer of the tx.

Same way any gov signed message is verified